kotsadm/pkg/policy/policy.go

```diff
@@ -35,32 +35,33 @@ func Must(p *Policy, err error) *Policy {
 
 func (p *Policy) Enforce(handler http.HandlerFunc) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
-		action, resource, err := p.execute(r)
+		action, resource, appSlug, err := p.execute(r)
 		if err != nil {
 			logger.Error(errors.Wrapf(err, "failed to execute policy template %q", p.resource))
 			w.WriteHeader(http.StatusInternalServerError)
 			return
 		}
-		if err := handlers.CheckAccessOrAbort(w, r, action, resource); err != nil {
+		if err := handlers.CheckAccessOrAbort(w, r, action, resource, appSlug); err != nil {
 			logger.Error(err)
 			return
 		}
 		handler(w, r)
 	}
 }
 
-func (p *Policy) execute(r *http.Request) (action, resource string, err error) {
+func (p *Policy) execute(r *http.Request) (action, resource, appSlug string, err error) {
 	vars := mux.Vars(r)
 	for _, fn := range p.varsGetterFns {
 		additionalVars, err := fn(vars)
 		if err != nil {
-			return action, resource, err
+			return action, resource, "", err
 		}
 		for key, val := range additionalVars {
 			vars[key] = val
 		}
 	}
 	var buf bytes.Buffer
 	err = p.resourceTemplate.Execute(&buf, vars)
-	return p.action, buf.String(), err
+	appSlug, _ = vars["appSlug"]
```

Comment:
won't this always overwrite the appSlug?

Reply:
i dont think this does what you think it does but also this wont work. i will fix it